src/main/resources/vm/script/datamanagement/uploadFunctions.vm

```diff
@@ -130,6 +130,34 @@ function uploadLfnFile {
         addToCache ${LFN} ${FILE}
     fi
 }
+function uploadShanoirFile {
+    local URI=$1
+
+    local token=`cat $shanoir_token_location`
+
+    local upload_url=`echo $URI | sed -r 's/^.*[?&]upload_url=([^&]*)(&.*)?$/\1/i'`
+    local fileName=`echo $URI | sed -r 's#^shanoir:/(//)?(.*/(.+))\?.*$#\3#i'`
+    local filePath=`echo $URI | sed -r 's#^shanoir:/(//)?([^/].*)\?.*$#\2#i'`
+   
+    local type=`echo $URI | sed -r 's/^.*[?&]type=([^&]*)(&.*)?$/\1/i'`
+    local md5=`echo $URI | sed -r 's/^.*[?&]md5=([^&]*)(&.*)?$/\1/i'`
+    
+
+    COMMAND(){ 
+        (echo -n '{"base64Content": "'; base64 ${fileName}; echo '", "type":"'; echo ${type}; echo '", "md5":"'; echo ${md5} ; echo '"}') | curl --write-out '%{http_code}' --request PUT "${upload_url}/${filePath}"  --header "Authorization: Bearer ${token}"  --header "Content-Type: application/carmin+json" --header 'Accept: application/json, text/plain, */*' -d @- --cacert /var/www/prod/shanoir-ng-nginx.pem
```

Comment:
Dear Axel, 

Can you please specify which CARMIN method you're referring to ? 
[https://app.swaggerhub.com/apis/CARMIN/carmin-common_api_for_research_medical_imaging_network/0.3.1#/default/uploadPath](url), this is the swagger I followed, the put method for upload.

With kind regards.

Reply:
I was thinking about the same method and sending the raw content instead of a base64 encoded in a "text" json file. Quoting the description : `All other content (with any content type) will be considered as a raw file and will override the existing path content. If the parent`. So if you send a zip file, with the zip binary content and `application/zip` in `content-type`, this should work and the payload will be smaller.